subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -810,6 +818,45 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			break;
 		}
 #endif
+#if defined(CONFIG_LOG_BACKEND_NET_USE_DHCPV4_OPTION)
+		case DHCPV4_OPTIONS_LOG_SERVER: {
+			struct sockaddr_in log_server;
+
+			/* Log server option may present 1 or more
+			 * addresses. Each 4 bytes in length. Log
+			 * servers should be listed in order
+			 * of preference.  Hence we choose the first
+			 * and skip the rest.
+			 */
+			if (length % 4 != 0U) {
+				NET_ERR("options_log_server, bad length");
+				return false;
+			}
+
+			(void)memset(&log_server, 0, sizeof(log_server));
+
+			if (net_pkt_read(pkt, log_server.sin_addr.s4_addr, 4) ||
+			    net_pkt_skip(pkt, length - 4U)) {
+				NET_ERR("options_log_server, short packet");
+				return false;
+			}
+
+			log_server.sin_family = AF_INET;
+			log_backend_net_set_ip((struct sockaddr *)&log_server);
+
+			const struct log_backend *backend = log_backend_net_get();
+
+			if (!log_backend_is_active(backend)) {
+
+				log_backend_init(backend);
+				log_backend_enable(backend, backend->cb->ctx, CONFIG_LOG_MAX_LEVEL);
+			}
```

Comment:
I don't really think we should have this here - the option should only update the log server address, not decide whether to activate the network log backed or not.

Reply:
This will now only be executed if `CONFIG_LOG_BACKEND_NET_AUTOSTART` is enabled, because if `CONFIG_LOG_BACKEND_NET_SERVER` is not set (empty string) during the init the net logging backend will be deactivated.

Reply:
Ok, it makes more sense now. At least the application which wants to have full control over the backend can disable the autostart feature.